e2e/cypress/integration/notifications/ignore_channel_mentions.js

```diff
@@ -0,0 +1,132 @@
+
+    // # Click on the notification preferences
+    cy.get('#channelNotificationPreferences').should('exist').click();
+
+    // # Click on the edit button for ignore channel mentions
+    cy.get('#ignoreChannelMentionsEdit').should('exist').click();
+
+    // # Click on selected option
+    cy.get(`#ignoreChannelMentions${toSet}`).should('exist').click();
+
+    // # Click on save to save the configuration
+    cy.get('#saveSetting').should('exist').click();
+
+    // * Assert that the option selected is reflected
+    cy.get('#ignoreChannelMentionsDesc').should('contain', toSet);
+
+    // # Click on the X button to close the modal
+    cy.get('#channelNotificationModalLabel').siblings('.close').click();
+}
+
+describe('CS15445 Join/leave messages', () => {
+    let testTeam;
+    let userB;
+
+    let channelA;
+    let channelB;
+
+    before(() => {
+        // # Login as new user and visit town-square
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+
+            // # Add 1 users
+            cy.apiCreateUser().then(({user: newUser}) => { // eslint-disable-line
+                userB = newUser;
+                cy.apiAddUserToTeam(testTeam.id, userB.id);
+            });
+
+            // # Create two channels
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelA = res.body;
+            });
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelB = res.body;
+            });
+
+            cy.apiLogin(user);
+        });
+    });
+
+    it('MM-T567 - Turn on Ignore mentions for @channel, @here and @all', () => {
+        cy.visit(`/${testTeam.name}/channels/${channelA.name}`);
+
+        // # Add users to channel
+        addNumberOfUsersToChannel(1);
+
+        cy.getLastPostId().then((id) => {
+            // * The system message should contain 'added to the channel by you'
+            cy.get(`#postMessageText_${id}`).should('contain', 'added to the channel by you');
+        });
+
+        setIgnoreMentions('On');
+
+        // # Go to a different channel
+        cy.visit(`/${testTeam.name}/channels/${channelB.name}`);
+
+        // # Post messages as another user on the first channel
+        cy.postMessageAs({sender: userB, message: '@all test', channelId: channelA.id});
+        cy.postMessageAs({sender: userB, message: '@channel test', channelId: channelA.id});
+        cy.postMessageAs({sender: userB, message: '@here test', channelId: channelA.id});
+
+        // * Assert the channel is unread with no mentions
+        cy.get(`#sidebarItem_${channelA.name}`).should('have.class', 'unread-title');
+        cy.get(`#sidebarItem_${channelA.name} > #unreadMentions`).should('not.exist');
+    });
+
+    it('MM-T568 - Turn off Ignore mentions for @channel, @here and @all', () => {
+        cy.visit(`/${testTeam.name}/channels/${channelA.name}`);
```

Comment:
Nit: this is repeated in both tests, maybe we can move it to the `before` block?

Reply:
In any case, it would be a before all, but even thought, I would prefer to keep it on each case. More cases may be added to this file, and they may need to go to different sites.

Reply:
Fair enough, makes sense! (although I still think it would be in the `before` block, not in the `beforeAll` :joy:)

Reply:
My bad, [the hooks](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Hooks) are `before` and `beforeEach`. And this would belong to `beforeEach`. I mixed all the names in my head!